frontend/src/components/pages/AdminDashboard/EditCheckInOrFoodRescueDescription.tsx

```diff
@@ -0,0 +1,164 @@
+      content.checkinUrl.length !== 0 &&
+      content.foodRescueDescription.length !== 0 &&
+      content.foodRescueUrl.length !== 0
+    ) {
+      const contentResponse = await ContentAPIClient.updateContent(
+        "1",
+        content,
+      );
+      setContent(contentResponse);
+      navigateToViewPage();
+    }
+  };
+
+  return (
+    <Container
+      alignContent="left"
+      variant="calendarContainer"
+      mt={["0.5rem", "2rem"]}
+    >
+      <Box>
+        <Box display="flex" justifyContent="right">
+          <CloseButton onClick={navigateToViewPage} />
+        </Box>
+        <Text textStyle="desktopHeader4" color="black.100">{`Edit ${
+          isCheckInView ? `check in` : `food rescue`
+        } description`}</Text>
+
+        <Text
+          textStyle={["mobileHeader4", "desktopSubtitle"]}
+          color="black.100"
+          mt={["34px", "35px"]}
+        >
+          Edit description
+        </Text>
+        <Textarea
+          mt={["17px", "42px"]}
+          value={
+            isCheckInView
+              ? content?.checkinDescription
+              : content?.foodRescueDescription
+          }
+          name="description"
+          onChange={(e) =>
+            content &&
+            setContent({
+              ...content,
+              ...(isCheckInView
+                ? { checkinDescription: e.target.value }
+                : { foodRescueDescription: e.target.value }),
+            })
+          }
+          onFocus={() => setInteractedWith(true)}
+          width={["100%", "478px"]}
+          height={["140px", "210px"]}
+          background="squash.100"
+          p="1.5rem"
+          isInvalid={
+            (isCheckInView &&
+              (!content?.checkinDescription ||
+                content?.checkinDescription.length === 0)) ||
+            (!isCheckInView &&
+              (!content?.foodRescueDescription ||
+                content?.foodRescueDescription.length === 0) &&
+              interactedWith)
+          }
+        />
+
+        <Text
+          textStyle={["mobileHeader4", "desktopSubtitle"]}
+          color="black.100"
+          mt={["36px", "56px"]}
+        >
+          Edit link
+        </Text>
+
+        <Input
+          mt={["17px", "42px"]}
+          name="url"
+          value={isCheckInView ? content?.checkinUrl : content?.foodRescueUrl}
+          onChange={(e) => {
+            if (content) {
+              setContent({
+                ...content,
+                ...(isCheckInView
+                  ? { checkinUrl: e.target.value }
+                  : { foodRescueUrl: e.target.value }),
+              });
+            }
+          }}
+          onFocus={() => setInteractedWith(true)}
+          width={["100%", "44rem"]}
+          height={["44px", "64px"]}
+          background="squash.100"
+          isInvalid={
+            (isCheckInView &&
+              (!content?.checkinUrl || content?.checkinUrl.length === 0)) ||
```

Comment:
oop, i dont think we need to check `!content?.checkinUrl`, `!content?.foodRescueUrl` for these

Reply:
dont they both check the same thing? `(!content?.checkinUrl || content?.checkinUrl.length === 0))`